src/notifications/NotificationsHandler.tsx

```diff
@@ -0,0 +1,331 @@
+    if (!store.getState().appState.walletReady) {
+      NotificationStorage.deferNotification(notification);
+      return;
+    }
+    performActionBasedOnOpenedNotificationType(notification);
+  };
+
+  const performActionBasedOnOpenedNotificationType = async (
+    notification: MinimalNotification
+  ) => {
+    const type = notification?.data?.type;
+
+    if (type === NotificationTypes.transaction) {
+      const untypedData = notification?.data;
+      if (
+        !untypedData?.address ||
+        !untypedData?.hash ||
+        !untypedData?.chain ||
+        !untypedData?.transaction_type
+      ) {
+        return;
+      }
+
+      // casting data payload to type that was agreed on with backend
+      const data = (notification.data as unknown) as TransactionNotificationData;
+
+      const { wallets, contacts } = syncedStateRef.current;
+      const { accountAddress, nativeCurrency } = store.getState().settings;
+
+      let walletAddress: string | null = accountAddress;
+      if (!isLowerCaseMatch(accountAddress, data.address)) {
+        walletAddress = await wallets.switchToWalletWithAddress(data.address);
+      }
+      if (!walletAddress) {
+        return;
+      }
+      Navigation.handleAction(Routes.PROFILE_SCREEN, {});
+      const zerionTransaction = store
+        .getState()
+        .data.transactions.find(tx =>
+          isLowerCaseMatch(ethereumUtils.getHash(tx) ?? '', data.hash)
+        );
+      if (zerionTransaction) {
+        showTransactionDetailsSheet(zerionTransaction, contacts, walletAddress);
+      } else {
+        const network = ethereumUtils.getNetworkFromChainId(
+          parseInt(data.chain, 10)
+        );
+        const provider = await getProviderForNetwork(network);
+        const rpcTransaction = await provider.getTransaction(data.hash);
+
+        const transactionConfirmed =
+          rpcTransaction?.blockNumber && rpcTransaction?.blockHash;
+        if (!transactionConfirmed) {
+          return;
+        }
+
+        const newTransactionDetails: NewTransactionOrAddCashTransaction = {
+          type: mapNotificationTransactionType(data.transaction_type),
+          network,
+          hash: rpcTransaction.hash,
+          status: TransactionStatus.unknown,
+          amount: rpcTransaction.value.toString(),
+          nonce: rpcTransaction.nonce,
+          from: rpcTransaction.from,
+          to: rpcTransaction.to ?? null,
+          asset: null,
+          gasLimit: rpcTransaction.gasLimit,
+          maxFeePerGas: rpcTransaction.maxFeePerGas,
+          maxPriorityFeePerGas: rpcTransaction.maxPriorityFeePerGas,
+          gasPrice: rpcTransaction.gasPrice,
+          data: rpcTransaction.data,
+        };
+
+        const parsedTransaction = await parseNewTransaction(
+          newTransactionDetails,
+          nativeCurrency
+        );
+        const resultTransaction = { ...parsedTransaction };
+        const minedAt = Math.floor(Date.now() / 1000);
```

Comment:
i was thinking about a case where the notification is pressed lets' say 2 hours after the tx was confirmed, then this time wouldn't be accurate by 2 hours, but at that time zerion would probably get the tx with the right time. we could request the block timestamp to get the accurate time  but is probably ok to have it this way